sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdServiceEndpoint.java

```diff
@@ -555,7 +555,7 @@ public IAddressResolver getAddressResolver() {
         }
 
         private void evict(final RntbdEndpoint endpoint) {
-            if (this.endpoints.remove(endpoint.remoteAddress().toString()) != null) {
+            if (this.endpoints.remove(endpoint.serverKey().getAuthority()) != null) {
```

Comment:
Can we always expect that serverKey().getAuthority() to match the URI physicalAddress.getAuthority()?

Reply:
If not mistaken:
`   this.serverKey = RntbdUtils.getServerKey(physicalAddress);
`
So it might be required to change at line 539 to match this expected computation:
`   return endpoints.computeIfAbsent(RntbdUtils.getServerKey(physicalAddress).getAuthority(), authority -> new RntbdServiceEndpoint(`

Reply:
Good observation , I noticed that as well,  RntbdUtils.getServerKey(physicalAddress) build the uri with scheme, host and port, ignoring rest.  So getAuthority for both return the same host and port.
It is covered in the test case.
However we could still create extra URI as you mentioned in 539, but it will be redundant object creation.

Reply:
Can you add a comment at line 558 with this explanation, why we choose this and avoid the creation of the redundant object? Otherwise from an outsider perspective it still looks like a potential mismatch :-)

Reply:
Is it worth abstracting it out as a contract/API, and leverage on related places for consistency?

Reply:
Can it then be limited to one implementation across or both needed?

Reply:
As mentioned above one implementation  will cost extra URI object creation on every request

Reply:
Talked offline , current approach is right